recipes-bsp/sgx/sgx-dcap_1.9.bb

```diff
@@ -0,0 +1,105 @@
+# Packages supported
+# sgx-dcap : SGX Data Center Attestation Package.
+
+### common ###
+
+SUMMARY = "Intel(R) SGX Data Center Attestation Primitives (DCAP) driver"
+LICENSE = "BSD & Intel-Redistributable-Binaries"
+LIC_FILES_CHKSUM = "file://License.txt;md5=66da782d3c2c41b9669340fb2704ac59"
+
+# Packages required to build.
+DEPENDS += "curl sgx-sdk-cross"
+
+# Source locations
+S = "${WORKDIR}/git"
+D_prebuilt_dcap = "${S}/QuoteGeneration"
+D_sgxssl="${S}/QuoteVerification/sgxssl"
+D_openssl="${S}/QuoteVerification/sgxssl/openssl_source"
+
+SRC_URI  = "git://git@github.com/intel/SGXDataCenterAttestationPrimitives.git"
+SRCREV = "98976322e8b58e23256355f5cf90b9e30e37d8c1"
+
+### configure prebuilt ###
+SRC_URI += "https://download.01.org/intel-sgx/sgx-dcap/1.9/linux/prebuilt_dcap_1.9.tar.gz;name=prebuilt_dcap;subdir=${D_prebuilt_dcap}"
+SRC_URI[prebuilt_dcap.sha256sum] = "8cd0249ee49dbfd589b257cd0fa14d374d01b6c210ca7d0a14e618a48bbdb82b"
+
+configure_prebuilt () {
+    :
+}
+
+### configure sgxssl ###
+S_sgxssl="${WORKDIR}/sgxssl"
+File_sgxssl="lin_2.11_1.1.1g"
+SRC_URI += "https://github.com/01org/intel-sgx-ssl/archive/${File_sgxssl}.zip;unpack=0;name=sgxssl;subdir=${WORKDIR}/sgxssl"
+SRC_URI[sgxssl.sha256sum] = "12828839c4555e0f5e88e86db090c995053d98d99091862c498fc55f379183fc"
+
+configure_sgxssl () {
+    # Copy intel-sgx-ssl to correct build location.
+    if [ ! -d "${D_sgxssl}/openssl_source" ]; then
+        if [ ! -d "${D_sgxssl}" ]; then
+            echo "Create ${D_sgxssl}"
+            mkdir -p "${D_sgxssl}"
+        fi
+        unzip "${S_sgxssl}/${File_sgxssl}.zip" -d "${D_sgxssl}-unzip"
+        mv ${D_sgxssl}-unzip/intel-sgx-ssl-${File_sgxssl}/* "${D_sgxssl}"
+        rmdir "${D_sgxssl}-unzip/intel-sgx-ssl-${File_sgxssl}"
+        rmdir "${D_sgxssl}-unzip"
+    fi
+}
+
+### configure openssl ###
+S_openssl="${WORKDIR}/openssl"
+File_openssl="openssl-1.1.1g"
+SRC_URI += "https://www.openssl.org/source/${File_openssl}.tar.gz;unpack=0;name=openssl;subdir=${S_openssl}"
+SRC_URI[openssl.sha256sum] = "ddb04774f1e32f0c49751e21b67216ac87852ceb056b75209af2443400636d46"
+
+configure_openssl () {
+    # Copy openssl to correct build location
+    if [ ! -f "${D_openssl}/${File_openssl}.tar.gz" ]; then
+        cp "${S_openssl}/${File_openssl}.tar.gz" "${D_openssl}"
+        if [ -d "${D_openssl}/${File_openssl}" ]; then
```

Comment:
don't we have to remove previous version before copying tar file over ?

Reply:
the tar file is copied and old version of openssl is removed (if exist) in line #62.

Reply:
Please use out of tree openssl, do not bundle your own copy, the latest is already 1.1.1k.  Just make a note that this recipe is only compatible with OpenSSL 1.1.X.

Reply:
Getting errors while compiling with 1.1.1K version

Reply:
got reply form sgx team as below:

Hi Antara,

SGX QvE is an Intel signed enclave for quote verification, it uses SgxSSL for some X.509 operations, and SgxSSL reuses/customized OpenSSL.
Here we build OpenSSL is target for SgxSSL, and the output is a trusted library for QvE.

So we cannot use system installed OpenSSL.  

Thanks,
HY

This component requires customized OpenSSL hence can not use out of tree openssl here.